gitpod-io/gitpod
# code user data sync

#### What it does

This PR exposes the code sync endpoint by the server component. The endpoint stores the metadata of resources in the DB and the content in the content service.

Changes in Code: https://github.com/gitpod-io/vscode/commit/f12434cc6ef580b85fbb457daaf8935670ddacc5

Difference between Code and Theia:
In Theia we operate user and workspace level extensions. In Code all extensions are user extensions, but they can be scoped by the machine (workspace origin in our case) via disabling the extension synching.

#### How to test

- smoke test Theia startup
- enable feature preview and Code
- start 2 workspaces
- change user settings (for instance color theme) and install extensions (for instance gitlens) in one workspace and check that they are reflected in another
- start another workspace and check that new user settings and extensions are applied
